project-base/storefront/helpers/errors/errorMessageMapper.ts

```diff
@@ -1,52 +1,36 @@
-        [ApplicationErrors['order-empty-cart']]: t('Cart is empty.'),
-        [ApplicationErrors['personal-data-request-type-invalid']]: t('Invalid request type.'),
-        [ApplicationErrors['blog-category-not-found']]: t('Category not found.'),
-        [ApplicationErrors['image-type-invalid']]: t('Invalid image type.'),
-        [ApplicationErrors['image-size-invalid']]: t('Invalid image size.'),
-        [ApplicationErrors['order-not-found']]: t('Order not found.'),
-        [ApplicationErrors['personal-data-hash-invalid']]: t('Invalid hash.'),
-        [ApplicationErrors['product-price-missing']]: t('Product price is missing.'),
-        [ApplicationErrors['store-not-found']]: t('Store not found.'),
-        [ApplicationErrors['product-not-found']]: t('Product not found.'),
-        [ApplicationErrors['handling-with-logged-customer-comparison']]: t('Product not found.'),
-        [ApplicationErrors['comparison-not-found']]: t('Comparison not found.'),
-        [ApplicationErrors['compared-item-not-found']]: t('Compared product not found.'),
-        [ApplicationErrors['compared-item-already-exists']]: t('Compared product is already compared.'),
+const getErrorMessageTranslationString = (errorCode: FlashMessageKeys, t: Translate): string | undefined => {
+    const ERROR_MESSAGES: Record<FlashMessageKeys, string | undefined> = {
+        default: t('Unknown error.'),
+        'cart-not-found': t('Cart not found.'),
+        'max-allowed-limit': t('Max allowed limit reached.'),
+        'packetery-address-id-invalid': t('Invalid Packetery address id.'),
+        'invalid-credentials': t('Invalid credentials.'),
+        'invalid-refresh-token': t('Invalid refresh token.'),
+        'order-emails-not-sent': t('Automatic order emails was not sent.'),
+        'order-empty-cart': t('Cart is empty.'),
+        'personal-data-request-type-invalid': t('Invalid request type.'),
+        'blog-category-not-found': t('Category not found.'),
+        'image-type-invalid': t('Invalid image type.'),
+        'image-size-invalid': t('Invalid image size.'),
+        'order-not-found': t('Order not found.'),
+        'personal-data-hash-invalid': t('Invalid hash.'),
+        'product-price-missing': t('Product price is missing.'),
+        'store-not-found': t('Store not found.'),
+        'product-not-found': t('Product not found.'),
+        'handling-with-logged-customer-comparison': t('Product not found.'),
+        'comparison-not-found': t('Comparison not found.'),
+        'compared-item-not-found': t('Compared product not found.'),
+        'compared-item-already-exists': t('Compared product is already compared.'),
     };
 
     return ERROR_MESSAGES[errorCode];
 };
 
-export const hasErrorMessage = (errorCode: string, t: Translate): boolean => {
-    return getErrorMessageTranslationString(errorCode as ApplicationErrorsWithoutIgnoredErrorsType, t) !== undefined;
-};
-
 export const getErrorMessage = (errorCode: string, t: Translate): string => {
```

Comment:
```suggestion
export const getErrorMessage = (errorCode: FlashMessageKeys & NoFlashMessageKeys, t: Translate): string => {
```

Reply:
I think what you meant is `FlashMessageKeys | NoFlashMessageKeys`, because `FlashMessageKeys & NoFlashMessageKeys` is `never`